diagnostics/zipkin/grpc_tracer.go

```diff
@@ -0,0 +1,141 @@
+/*
+* Copyright 2021 Layotto Authors
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+ */
+
+package zipkin
+
+import (
+	"context"
+	"time"
+
+	"mosn.io/layotto/diagnostics/grpc"
+
+	"github.com/openzipkin/zipkin-go"
+	reporterhttp "github.com/openzipkin/zipkin-go/reporter/http"
+	"mosn.io/api"
+	"mosn.io/mosn/pkg/types"
+	"mosn.io/pkg/log"
+
+	ltrace "mosn.io/layotto/components/trace"
+)
+
+const (
+	service_name       = "service_name"
+	reporter_endpoint  = "reporter_endpoint"
+	recorder_host_post = "recorder_host_post"
+	configs            = "config"
+
+	defaultReporterEndpoint = "http://127.0.0.1:9411/api/v2/spans"
+	defaultServiceName      = "layotto"
+	defaultRecorderHostPost = "127.0.0.1:9000"
+)
+
+type grpcZipTracer struct {
+	*zipkin.Tracer
+}
+
+type grpcZipSpan struct {
+	*ltrace.Span
+	tracer *grpcZipTracer
+	ctx    context.Context
+	span   zipkin.Span
+}
+
+func NewGrpcZipTracer(traceCfg map[string]interface{}) (api.Tracer, error) {
+	reporter := reporterhttp.NewReporter(getReporterEndpoint(traceCfg))
+
+	endpoint, err := zipkin.NewEndpoint(getServerName(traceCfg), getRecorderHostPort(traceCfg))
+	if err != nil {
+		log.DefaultLogger.Errorf("[layotto] [zipkin] [tracer] unable to create zipkin reporter endpoint")
+		return nil, err
+	}
+
+	tracer, err := zipkin.NewTracer(reporter, zipkin.WithLocalEndpoint(endpoint), zipkin.WithTraceID128Bit(true))
+	if err != nil {
+		log.DefaultLogger.Errorf("[layotto] [zipkin] [tracer] cannot initialize zipkin Tracer")
+		return nil, err
+	}
+
+	log.DefaultLogger.Infof("[layotto] [zipkin] [tracer] create success")
+
+	return &grpcZipTracer{
+		tracer,
+	}, nil
+}
+
+func getRecorderHostPort(traceCfg map[string]interface{}) string {
+	if cfg, ok := traceCfg[configs]; ok {
+		recorderHostPort := cfg.(map[string]interface{})
+		if point, ok := recorderHostPort[recorder_host_post]; ok {
+			return point.(string)
+		}
+	}
+
+	return defaultRecorderHostPost
```

Comment:
Since all the fields are marked as "required" in the doc, shall we return error when there is no config?
![image](https://user-images.githubusercontent.com/26001097/174757927-4a03e02a-253a-4c7c-804d-a8d6f0a0fb51.png)